src/event.rs

```diff
@@ -239,94 +221,157 @@ pub enum WindowEvent<'a> {
+    ThemeChanged(Theme),
+}
 
-    /// Motion on some analog axis. May report data redundant to other, more specific events.
-    AxisMotion {
-        device_id: DeviceId,
-        axis: AxisId,
-        value: f64,
-    },
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct Vector<T> {
+    pub x: T,
+    pub y: T,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct KeyPress {
+    pub(crate) logical_key: Option<LogicalKey>,
+    pub(crate) scan_code: u32,
+    pub(crate) is_down: bool,
+    pub(crate) repeat_count: u32,
+    pub(crate) is_synthetic: bool,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct RawKeyPress {
+    pub(crate) logical_key: Option<LogicalKey>,
+    pub(crate) scan_code: u32,
+    pub(crate) is_down: bool,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct PointerPress {
+    pub(crate) button: u8,
+    pub(crate) is_down: bool,
+    pub(crate) click_count: u32,
+}
 
-    /// Touch event has been received
-    Touch(Touch),
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct RawPointerPress {
+    pub(crate) button: u8,
+    pub(crate) is_down: bool,
+}
 
+impl KeyPress {
+    pub fn logical_key(&self) -> Option<LogicalKey> {
+        self.logical_key
+    }
+    pub fn scan_code(&self) -> u32 {
+        self.scan_code
+    }
+    pub fn is_down(&self) -> bool {
+        self.is_down
+    }
+    /// The number of repeat events that have been generated from the user holding the key down. Is
+    /// `0` for the first event, and increments by `1` for every further event.
+    ///
+    /// TODO PHRASE BETTER
+    pub fn repeat_count(&self) -> u32 {
+        self.repeat_count
+    }
+    /// If set, the event was generated synthetically by winit
+    /// in one of the following circumstances:
+    ///
+    /// * Synthetic key press events are generated for all keys pressed
+    ///   when a window gains focus. Likewise, synthetic key release events
+    ///   are generated for all keys pressed when a window goes out of focus.
+    ///   ***Currently, this is only functional on X11 and Windows***
+    ///
+    /// Otherwise, this value is always `false`.
+    pub fn is_synthetic(&self) -> bool {
+        self.is_synthetic
+    }
+}
+
+impl RawKeyPress {
+    pub fn logical_key(&self) -> Option<LogicalKey> {
+        self.logical_key
+    }
+    pub fn scan_code(&self) -> u32 {
+        self.scan_code
+    }
+    pub fn is_down(&self) -> bool {
+        self.is_down
+    }
+}
+
+impl PointerPress {
+    pub fn button(&self) -> u8 {
+        self.button
+    }
+    pub fn is_down(&self) -> bool {
+        self.is_down
+    }
+    /// The number of clicks the user has made in the same spot. Is `1` for the first click, `2`
+    /// for the second click, etc.
+    pub fn click_count(&self) -> u32 {
```

Comment:
I'm guessing this is intended to be a generalization of double-click detection, but from the description I'd expect it to count the number of times the mouse was clicked without being moved regardless of how much time passed between those clicks, which would be very strange.

Reply:
You're right on both counts. I've updated the documentation to make the intent more clear.